### DISCUSSION TITLE
Balance between optuna and early stopping (Deep Learning)

### DISCUSSION THREAD

**Discussion by ken-take-it-so-so** [optuna/optuna#4194] [Q&A] [answered]
Hello. This is the first time to use optuna.  
Does optuna have below feature? I write a scenario.  
- I want to apply early stopping to prevent overfitting every optuna trials.  
- Early stopping is judged by validation loss, but optuna  explores to minimize validation MSE.  
- Patience of early stopping is 20. If the best number of epochs is 80, training will be done 100 epochs.
- In this case, does optuna evaluate using 80 epochs validation MSE, not 100 epochs validation MSE?  

I would appreciate your cooperation.

**Comment by nzw0301** [comment, marked as answer]
No, Optuna doesn't have an early-stopping feature like https://keras.io/api/callbacks/early_stopping/ for each trial. It should be done in each optimisation function.

**Reply by ken-take-it-so-so** [reply]
Thank you for your response! ありがとうございます．
